modules/configuration/src/main/java/org/apache/ignite/internal/configuration/hocon/HoconObjectConfigurationSource.java

```diff
@@ -87,6 +99,12 @@ public void descend(ConstructableTreeNode node) {
     }
 
     private void parseConfigEntry(String key, ConfigValue hoconCfgValue, ConstructableTreeNode node) {
+        for (Pattern deletedPrefix : deletedPrefixes) {
+            if (deletedPrefix.matcher(join(appendKey(path, key))).matches()) {
+                return;
+            }
+        }
```

Comment:
Such a pattern is repeated multiple times. Maybe we should introduce an interface, this way we'll be able to replace its implementation with a more efficient one in the future!